### PR TITLE
Session Timeout

### PR DESCRIPTION
**Pull request by ChazUK**
Adding an Idle Service that will log the user out after `x` amount of seconds in inactivity

The service also ping's every `x` amount of seconds which can be used to refresh tokens.

### CODE REVIEW

**Comment by ChazUK** [comment]
[Expiry Session - Log Out](https://trello.com/c/JLzv3aIF/421-expiry-session-log-out)

File: src/app/features/login/login.component.ts [hunk 1]
```diff
@@ -89,6 +91,15 @@ export class LoginComponent implements OnInit, OnDestroy {
 
           const token = response.headers.get('authorization');
           this.authService.authorise(token);
+
+          this.idleService.init(60, 600);
```
Comment (wozzer72) [inline src/app/features/login/login.component.ts]: what do these hard coded numbers mean?

File: src/app/features/login/login.component.ts [hunk 2]
```diff
@@ -89,6 +91,15 @@ export class LoginComponent implements OnInit, OnDestroy {
...
+          this.idleService.start();
+
+          this.idleService.ping$.subscribe(() => console.log('REFRESH BEARER TOKEN'));
```
Comment (wozzer72) [inline src/app/features/login/login.component.ts]: I take it this is where you'll call the new API endpoint?

It needs to be able to handle 403 errors - failed refresh

Comment (wozzer72) [reply src/app/features/login/login.component.ts]: It needs to replace the existing token.

File: src/app/features/login/login.component.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/features/login/login.component.ts]: 60 seconds to Ping, 600 seconds to timeout. I can put them as constants in a file somewhere but they're still going to be hardcoded

File: src/app/features/login/login.component.ts (hunk 2 above)
Comment (ChazUK) [reply src/app/features/login/login.component.ts]: Yep, haven't implemented that yet, just a placeholder

File: src/app/features/login/login.component.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/features/login/login.component.ts]: Updated to Constants at the top of the file

**Pull request merged**